src/components/views/elements/AppPermission.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import url from 'url';
+
+export default class AppPermission extends React.Component {
+    constructor(props) {
+        super(props);
+
+        const curlBase = this.getCurlBase();
+        this.state = {
+            curlBase: curlBase,
+        };
+    }
+
+    // Return string representation of content URL without query parameters
+    getCurlBase() {
+        const wurl = url.parse(this.props.url);
+        let curl;
+        let curlString;
+
+        const searchParams = new URLSearchParams(wurl.search);
+
+        if(searchParams && searchParams.get('url')) {
```

Comment:
Hmm... this might just be a case of renaming this variable (to something like curlBase), as I have done elsewhere in the function. This is only intended to be a display label specific to the widget type / content. I've got some examples, so maybe it is best to discuss this in person on Monday? Leaving this as it is for now.